<launch>
	<arg name="veh" doc="Name of vehicle. ex: megaman"/>
	<arg name="local" default="true" doc="true for running everything you possibly can on laptop. false for everything running on vehicle."/>
	<arg name="config" default="baseline" doc="Specify a config."/>
	<arg name="param_file_name" default="default" doc="Specify a param file. ex:megaman." />
	<arg name="raw" default="false" doc="If set to true, also publish raw image using decoder."/> 
	<arg name="live" default="true" doc="if set to true, run the camera and do live detection. "/>
	<arg name="camera" default="true" doc="If set to false, won't launch the camera. This is for easier relaunching of the line_detector_node only "/>
        <arg name="verbose" default="true" />
	<arg name="camera_topic" default="camera_node"/>
        <arg name="line_detector_param_file_name" default='default' />
        <arg name="drive" default="true"/>
	<!-- Camera -->
	<include if="$(arg camera)" file="$(find duckietown)/launch/camera.launch">
		<arg name="veh" value="$(arg veh)"/>
		<arg name="local" value="$(arg local)"/>
		<arg name="config" value="$(arg config)"/>
		<arg name="param_file_name" value="$(arg param_file_name)"/>
		<arg name="raw" value="$(arg raw)"/>
		<arg name="cam_info" value="true"/> <!-- this should possible change later when we take logs without camera info but let's leave for now -->
		<arg name="live" value="$(arg live)"/>

	</include>

	<!-- Line Detector -->
	<remap from="line_detector_node/image" to="camera_node/image/compressed"/>
	<include file="$(find ino_car)/launch/ino_line_detector_node.launch">
	    <arg name="veh" value="$(arg veh)"/>
	    <arg name="local" value="$(arg local)"/>

	</include>

        <!-- ground projection-->	
	<include file="$(find ino_car)/launch/ino_ground_projection_node.launch">
		<arg name="veh" value="$(arg veh)"/>
		 <arg name="local" value="$(arg local)"/>
		 <arg name="config" value="$(arg config)"/>
		 <arg name="param_file_name" value="$(arg param_file_name)"/>
		 <arg name="camera_topic" value="$(arg camera_topic)"/>
	</include>

	<!-- Lane controller -->
        <remap from="lane_controller_node/lane_pose" to="ground_projection/goal_point"/>
        <include file="$(find ino_car)/launch/ino_lane_controller_node.launch">
        	<arg name="veh" value="$(arg veh)"/>
        	<arg name="local" value="$(arg local)"/>
        	<arg name="config" value="$(arg config)"/>
        	<arg name="param_file_name" value="$(arg param_file_name)"/>        
    </include>

        <remap from="base_ino_control/car_cmd" to="lane_controller_node/car_cmd"/>
        <include file="$(find ino_car)/launch/ino_kb.launch">
        	<arg name="veh" value="$(arg veh)"/>
        	<arg name="local" value="$(arg local)"/>
        	<arg name="config" value="$(arg config)"/>
        	<arg name="param_file_name" value="$(arg param_file_name)"/>        
    </include>
</launch>
